eddies/github-webhook-cloud-function
# Show PR review status

**Adds support for Review custom field**
* * *

# What does this Pull Request do?
Show a PR's review status, either:
* comma-separated list of requested reviewers,
* a ⭕ if a reviewer has requested changes
* a ✅ if a reviewer has approved the PR

Swaps the now-deprecated @google-cloud/functions-emulator for
@google-cloud/functions-framework

Swaps the runtime from node8 to node10

Adds nodemon for "hot-reload" support for local development with
`npm-start`

Bumps devDependencies (eslint-plugin-import, jest)

# How should this be tested?
```console
$ npm install
$ npm test
$ export GITHUB_SECRET=<your github secret>
$ export TRELLO_API_KEY=<your key here>
$ export TRELLO_TOKEN=<your token here>
$ npm start
```

Issue some curl requests against your local dev instance, e.g.:

```console
$ export trelloCard="abc12def/7-foo"
$ export pr="https://github.com/github/linguist/pull/11"
$ payload='{"action":"review_requested","pull_request":{"requested_reviewers":[{"login":"alice"}],"html_url":"'${pr}'","head":{"ref":"'${trelloCard}'"}}}'
$ signature=$(node -p "github = require('./src/util/github');github.sign('${payload}')")
$ curl -H "X-Hub-Signature: ${signature}" \
-H "X-GitHub-Event: pull_request" -H "X-GitHub-Delivery: 123" \
-H "Content-Type: application/json" -d ${payload} \
http://localhost:8080/
```

(`trelloCard` needs to be a card you have write access to).

Verify that the Trello Card has the requested reviewers you specified.

# Checklist
